src/varname.jl

```diff
@@ -104,78 +105,83 @@ julia> getsym(@varname(y))
 getsym(vn::VarName{sym}) where {sym} = sym
 
 """
-    getlens(vn::VarName)
+    getoptic(vn::VarName)
 
-Return the lens of the Julia variable used to generate `vn`.
+Return the optic of the Julia variable used to generate `vn`.
 
 ## Examples
 
 ```jldoctest
-julia> getlens(@varname(x[1][2:3]))
-(@lens _[1][2:3])
+julia> getoptic(@varname(x[1][2:3]))
+(@o _[1][2:3])
 
-julia> getlens(@varname(y))
-(@lens _)
+julia> getoptic(@varname(y))
+identity (generic function with 1 method)
 ```
 """
-getlens(vn::VarName) = vn.lens
-
+getoptic(vn::VarName) = vn.optic
 
 """
     get(obj, vn::VarName{sym})
 
-Alias for `get(obj, PropertyLens{sym}() ∘ getlens(vn))`.
+Alias for `getoptic(vn)(obj)`.
 """
-function Setfield.get(obj, vn::VarName{sym}) where {sym}
-    return Setfield.get(obj, PropertyLens{sym}() ∘ getlens(vn))
+function Base.get(obj, vn::VarName{sym}) where {sym}
+    return getoptic(vn)(obj)
 end
 
 """
     set(obj, vn::VarName{sym}, value)
 
-Alias for `set(obj, PropertyLens{sym}() ∘ getlens(vn), value)`.
+Alias for `set(obj, PropertyLens{sym}() ∘ getoptic(vn), value)`.
 """
-function Setfield.set(obj, vn::VarName{sym}, value) where {sym}
-    return Setfield.set(obj, PropertyLens{sym}() ∘ getlens(vn), value)
+function Accessors.set(obj, vn::VarName{sym}, value) where {sym}
+    return Accessors.set(obj, PropertyLens{sym}() ∘ getoptic(vn), value)
 end
 
 
-Base.hash(vn::VarName, h::UInt) = hash((getsym(vn), getlens(vn)), h)
+Base.hash(vn::VarName, h::UInt) = hash((getsym(vn), getoptic(vn)), h)
 function Base.:(==)(x::VarName, y::VarName)
-    return getsym(x) == getsym(y) && getlens(x) == getlens(y)
+    return getsym(x) == getsym(y) && getoptic(x) == getoptic(y)
 end
 
 # Allow compositions with lenses.
-function Base.:∘(vn::VarName{sym,<:Lens}, lens::Lens) where {sym}
-    return VarName{sym}(getlens(vn) ∘ lens)
+function Base.:∘(vn::VarName{sym,T}, optic) where {sym,T}
+    return VarName{sym}(getoptic(vn) ⨟ optic)
+end
+function Base.:∘(vn::VarName{sym,typeof(identity)}, optic) where {sym}
+    return VarName{sym}(optic)
 end
```

Comment:
Hmm, I'm a bit uncertain about this.

We implemented this originally to replicate the composition behavior of Setfield.jl, but now this reversed and so it seems a bit strange to not also reverse this here?

Reply:
The thing is, we are making `VarName` act as, what was before, a `PropertyLens`, in which case it _does_ make sense to consider the composition IMO. We are still doing this, no?